### PR TITLE
ymf271.cpp: Fix timer A period, Add notes for timer

### PR DESCRIPTION
**Pull request by cam900**

### CODE REVIEW

**Comment by ValleyBell** [comment]
The original comment said

> According to Yamaha's documentation, this sets timer A upper 2 bits

I think it would be nice to keep that note and include an *additional* note that the documentation is wrong here.

File: src/devices/sound/ymf271.cpp [hunk 1]
```diff
@@ -1371,14 +1371,13 @@ void ymf271_device::ymf271_write_timer(uint8_t address, uint8_t data)
...
 		{
 			case 0x10:
-				m_timerA = data;
+				m_timerA = (m_timerA & 0x003) | (uint32_t(data) << 2); // High 8 bit of Timer A period
 				break;
 
 			case 0x11:
-				// According to Yamaha's documentation, this sets timer A upper 2 bits
-				// (it says timer A is 10 bits). But, PCB audio recordings proves
-				// otherwise: it doesn't affect timer A frequency. (see ms32.c tetrisp)
```
Comment (angelosa) [inline src/devices/sound/ymf271.cpp:1380]: Still not sure about who told you that removing actual concrete examples from commentary is a good idea.

Comment (happppp) [reply src/devices/sound/ymf271.cpp:1380]: It just means Yamaha's documentation had it the wrong way around. If the fix is correct, the note doesn't apply anymore.

Comment (ValleyBell) [reply src/devices/sound/ymf271.cpp:1380]: Yes, but if the documentation is wrong, it should be noted to avoid future confusion.

Comment (DavidHaywood) [reply src/devices/sound/ymf271.cpp:1380]: yeah I'm inclined to agree, it's worth starting when otherwise 'trusted' documentation is incorrect.

Comment (angelosa) [reply src/devices/sound/ymf271.cpp:1380]: The old note mentions PCB recordings proving that there's some inconsistency with this register, that has been wiped out with current commentary. 

Also I've tested out a few entries that playback BGM on attract (rfjet, gametngk, desertwr) and audio analyzer sports subtle and slight differences in behaviour. So subtle that I can't pinpoint exactly what lacking better audio program analyzer atm than Audacity (beat tempo?). 

Bottom line: I'm honestly curious about the rationale, how it has been tested etc.

Comment (cam900) [reply src/devices/sound/ymf271.cpp:1380]: It's guesswork for Similar timer behavior with other Yamaha FM chips; and sound driver codes at some games/machines are write here for low(not high) 2 bits of timer A period, it's why subtle/slight differences compares to previous behaviour.

There's more similarity point with OPX between other FM chips? needs to research and verifications. (ex: Detune, Timer B free-running *16 multiplier, etc...)

Comment (happppp) [reply src/devices/sound/ymf271.cpp:1380]: I tested your change against a tetrisp pcb recording from youtube, one song that writes 3 to this register, one song that writes 0. Both match correctly with the new change.
With the previous implementation, the song that wrote 3 gets out of sync. The old note about tetrisp is invalid (IIRC i'm the one that wrote that ;p)
